File: game/cave/talking_before_climb.py
import pygame

from animation import Animation
from cave.cave_resources import CaveResources
from cave.climbing import Climbing
from config import Config
from phone_events import PhoneEvents
from state import State


class TalkingBeforeClimb(State):
    def process_events(self, phone_events: PhoneEvents):
        if not Config.ARGENTINE_VERSION and self.one_shot(4, "TrappeGrin"):
                pygame.mixer.Sound.play(CaveResources.trappe_grin)

        if self.get_frame_index() >= len(CaveResources.sync_hugo_start):
            return Climbing
        return None

    def render(self, screen):
        screen.blit(Animation.get_sync_frame(CaveResources.talks, CaveResources.sync_hugo_start, self.get_frame_index()), (0, 0))

    def on_enter(self) -> None:
        super().on_enter()
        pygame.mixer.Sound.play(CaveResources.trappe_snak)

File: game-web/src/cave/talkingBeforeClimb.ts
/** Port of game/cave/talking_before_climb.py */

import { State, type StateClass } from '../state';
import { Animation } from '../animation';
import { AudioHelper } from '../audioHelper';
import { CaveResources } from './caveResources';
import { Climbing } from './climbing';
import { Config } from '../config';
import type { PhoneEvents } from '../phoneEvents';

export class TalkingBeforeClimb extends State {
  processEvents(_events: PhoneEvents): StateClass | null {
    if (!Config.ARGENTINE_VERSION && this.oneShot(4, 'TrappeGrin')) {
      AudioHelper.play(CaveResources.trappeGrin);
    }
    if (this.getFrameIndex() >= CaveResources.syncHugoStart.length) return Climbing;
    return null;
  }

  onEnter(): void {
    super.onEnter();
    AudioHelper.play(CaveResources.trappeSnak);
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    ctx.drawImage(Animation.getSyncFrame(CaveResources.talks, CaveResources.syncHugoStart, this.getFrameIndex()), 0, 0);
  }
}
